src/IO/VTK.cpp

```diff
@@ -695,12 +695,21 @@ void VTK::writeData( ){
     int                 length;
     char*               buffer ;
 
-    str.open( m_fh.getPath( ), std::ios::in | std::ios::out ) ;
+
+    str.open( m_fh.getPath( ), std::ios::in | std::ios::out | std::ios::binary ) ;
+    // note for developers. Opening this ascii header in binary is necessary to
+    // make seekg/tellg methods (used after) works properly. Ascii opening does not
+    // guarantee a correct stream moving/re-positioning, because of automatic jump of escape
+    // characters \n (unix) \r\n (windows). Binary opening does the trick.
     if (!str.is_open()) {
         throw std::runtime_error("Cannot create file \"" + m_fh.getName() + "\"" + " inside the directory \"" + m_fh.getDirectory() + "\"");
     }
+    //Position the input stream at the beginning
+    str.seekg(0, std::ios::beg);
 
-    { // Write Ascii
+    { // Write Ascii - It may seem disturbing to write ascii on a binary opened
+      // file but genericIO::flushBinary (reinterpret_cast of chars into chars)
+      // should ensure no surprises into the final file.
 
```

Comment:
"Should ensure no surprises" is not very reassuring. If we are sure this works let's say that in the comment, otherwise it's better to do some more tests to make sure there are no actual surprises.

Reply:
Ironically, this was the most stressed part of my tests. I've modified the comment